lib/ClangImporter/SwiftLookupTable.h

```diff
@@ -133,6 +133,21 @@ class EffectiveClangContext {
     assert(getKind() == UnresolvedContext);
     return StringRef(Unresolved.Data, UnresolvedLength);
   }
+
+  /// Compares two EffectiveClangContexts without resolving unresolved names.
+  friend bool operator==(const EffectiveClangContext &lhs,
+                         const EffectiveClangContext &rhs) {
+    if (lhs.getKind() != rhs.getKind())
```

Comment:
As mentioned earlier, this will yield a false negative when one is resolved and not the other. Not sure if that's a scenario which is important to you. Might be worth having a function with a name rather than ==.